src/tests/ftest/performance/mdtest_hard_pil4dfs.yaml

```diff
@@ -0,0 +1,62 @@
+hosts:
+  test_servers: 2
+  test_clients: 2
+timeout: 300
+server_config:
+  name: daos_server
+  control_log_mask: INFO
+  engines_per_host: 2
+  engines:
+    0:
+      pinned_numa_node: 0
+      nr_xs_helpers: 1
+      fabric_iface: ib0
+      fabric_iface_port: 31317
+      log_file: daos_server0.log
+      log_mask: ERR
+      storage: auto
+    1:
+      pinned_numa_node: 1
+      nr_xs_helpers: 1
+      fabric_iface: ib1
+      fabric_iface_port: 31417
+      log_file: daos_server1.log
+      log_mask: ERR
+      storage: auto
+pool:
+  size: 95%
+  control_method: dmg
+  properties: ec_cell_sz:128KiB
+container:
+  type: POSIX
+  control_method: daos
+mdtest: &mdtest_base
+  client_processes:
+    ppn: 32
+  env_vars:
+    - D_LOG_MASK=ERR
+  test_dir: "/"
+  manager: "MPICH"
+  flags: "-C -T -r -F -P -G 27 -N 1 -Y -v -t -X -E"
+  api: "POSIX"
+  read_bytes: 3901
+  write_bytes: 3901
+  num_of_files_dirs: 100000000
+  stonewall_timer: 30
+  stonewall_statusfile: "/var/tmp/daos_testing/stoneWallingStatusFile"
+  dfs_destroy: false
+mdtest_pil4dfs_s1:
+  <<: *mdtest_base
+  dfs_oclass: S1
+  dfs_dir_oclass: SX
+  dfs_chunk: 1MiB
+mdtest_pil4dfs_ec_16p2g1:
+  <<: *mdtest_base
+  dfs_oclass: EC_16P2G1
+  dfs_dir_oclass: RP_3GX
```

Comment:
If the rest of the config is the same as `ftest/performance/mdtest_hard.yaml`, you could just add your new variants to the existing `ftest/performance/mdtest_hard.[py,yaml]`

Reply:
Other than "mdtest_pil4dfs_s1:" and "mdtest_pil4dfs_ec_16p2g1:", api: "POSIX" is different from api: "DFS" in mdtest_hard.yaml.

Reply:
If you're inclined...
If it's just "API: ", you could move the "API: DFS" to under each inherited namespace in mdtest_hard.yaml, then add your variants. I.e.
```
mdtest: &mdtest_base
  api: "DFS"  <-- remove this
mdtest_dfs_s1:
  <<: *mdtest_base
  api: "DFS"  <-- add it here
mdtest_dfs_ec_16p2g1:
  <<: *mdtest_base
  api: "DFS"  <-- and here
```
And then add your new variants.

Reply:
As you suggested, I merged ftest/performance/mdtest_hard_pil4dfs.[py,yaml] into mdtest_hard.[py,yaml]